src/ImageSharp/Color/Color.cs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Buffers.Binary;
+using System.Globalization;
+using System.Numerics;
+using System.Runtime.InteropServices;
+
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp
+{
+    /// <summary>
+    /// Represents a color value that is convertible to any <see cref="IPixel{TSelf}"/> type.
+    /// </summary>
+    /// <remarks>
+    /// The internal representation and layout of this structure is hidden by intention.
+    /// It's not serializable, and it should not be considered as part of a contract.
+    /// Unlike System.Drawing.Color, <see cref="Color"/> has to be converted to a specific pixel value
+    /// to query the color components.
+    /// </remarks>
+    public readonly partial struct Color : IEquatable<Color>
+    {
+        private readonly Rgba64 data;
```

Comment:
Are we not going to lose accuracy when converting to/from `RgbaVector` here due to rounding. Do we even need the vector type?

Reply:
I also keep thinking on this.
We never actually convert a `Color` from a pixel of an `Image<T>`, so there is no practical need for that scenario probably.